tests/test_log.py

```diff
@@ -114,3 +115,47 @@ def test_logging_constructor():
             p.poll(timeout=0.5)
 
         print('%s: %s: %d log messages seen' % (how, f.name, f.cnt))
+
+
+def test_producer_logger_logging_in_given_format():
+    """Test that asserts that logging is working by matching the log message"""
+
+    stringBuffer = StringIO()
+    logger = logging.getLogger('Producer logger')
+    logger.setLevel(logging.DEBUG)
+    handler = logging.StreamHandler(stringBuffer)
+    handler.setFormatter(logging.Formatter('The logger is - %(name)s'))
+    logger.addHandler(handler)
+
+    p = confluent_kafka.Producer(
+        {"bootstrap.servers": "test", "logger": logger, "debug": "msg"})
+    val = 1
+    while val > 0:
+        val = p.flush()
+    logMessage = stringBuffer.getvalue().strip()
+    stringBuffer.close()
+
+    assert logMessage.startswith('The logger is - Producer logger')
+    assert logMessage.endswith('The logger is - Producer logger')
```

Comment:
rather than check ends with, maybe check for a substring that you know for sure is going to be in the librdkafka emitted log.

Reply:
Checking the log is for the INIT call.